adapters/beachfront/beachfront.go

```diff
@@ -290,121 +344,112 @@ func getBannerRequest(req *openrtb.BidRequest) (BeachfrontBannerRequest, []error
+so each video imp is a call to the endpoint.
+*/
+func getVideoRequests(request *openrtb.BidRequest) ([]beachfrontVideoRequest, []error) {
+	var beachfrontReqs = make([]beachfrontVideoRequest, 0)
+	var errs = make([]error, 0, len(request.Imp))
 
-	if req.App != nil {
-		if req.App.Domain != "" {
-			beachfrontReq.Site.Domain = req.App.Domain
-			beachfrontReq.Site.Page = req.App.ID
-		}
-	} else {
-		if req.Site.Page != "" {
-			if req.Site.Domain == "" {
-				if strings.Contains(req.Site.Page, "//") {
-					// Remove protocol if exists
-					beachfrontReq.Site.Domain = strings.Split(req.Site.Page, "//")[1]
-				}
-				if strings.Contains(beachfrontReq.Site.Domain, "/") {
-					// Drop everything after the first "/"
-					beachfrontReq.Site.Domain = strings.Split(beachfrontReq.Site.Domain, "/")[0]
-				}
-			} else {
-				beachfrontReq.Site.Domain = req.Site.Domain
-			}
-			beachfrontReq.Site.Page = req.Site.Page
-		}
-	}
+	for i := 0; i < len(request.Imp); i++ {
+		r := newBeachfrontVideoRequest()
+		r.Imp = append(r.Imp, beachfrontVideoImp{})
 
-	/*
-		The req could contain banner,audio,native and video imps when It arrives here. I am only
-		interested in video
+		// The backend can take an array of Imps, but will only ever return
+		// a single seatbid with a single nurl, so videoIndex should always be 0,
+		// making one Imp per request.
+		var videoIndex = 0
 
-		The beachfront video endpoint is only capable of returning a single nurl and price, wrapped in
-		an openrtb format, so even though I'm building a request here that will include multiple video
-		impressions, only a single URL will be returned. Hopefully the beachfront endpoint can be modified
-		in the future to return multiple video ads
+		beachfrontExt, err := getBeachfrontExtension(request.Imp[i])
 
-	*/
-	for _, imp := range req.Imp {
-		if imp.Video != nil {
-			beachfrontReq.ID = req.ID
+		if err == nil {
+			r.Imp[videoIndex].Bidfloor = beachfrontExt.BidFloor
 
-			beachfrontReq.Imp = append(beachfrontReq.Imp, BeachfrontVideoImp{})
-			videoImpsIndex = len(beachfrontReq.Imp) - 1
+			if beachfrontExt.AppId != "" {
+				r.AppId = beachfrontExt.AppId
+			} else {
+				r.AppId = beachfrontExt.AppIds.Video
+			}
+		} else {
+			// Failed to extract the beachfrontExt, so this request is junk.
+			errs = append(errs, err)
+			continue
+		}
 
-			beachfrontReq.Imp[videoImpsIndex].Video.H = imp.Video.H
-			beachfrontReq.Imp[videoImpsIndex].Video.W = imp.Video.W
+		if request.Imp[i].Video.H != 0 && request.Imp[i].Video.W != 0 {
+			r.Imp[videoIndex].Video.W = request.Imp[i].Video.W
+			r.Imp[videoIndex].Video.H = request.Imp[i].Video.H
+		} else {
+			r.Imp[videoIndex].Video.W = DefaultVideoWidth
+			r.Imp[videoIndex].Video.H = DefaultVideoHeight
+		}
 
-			var bidderExt adapters.ExtImpBidder
-			if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-				errs = append(errs, err)
-				continue
+		if request.App != nil {
```

Comment:
I'm not saying it _can't_, but it gets really tricky since such a method would need to be able to handle both beachfrontVideoRequest enums and beachfrontBannerRequest enums as arguments and return values. I think the spirit of your comment is to avoid duplicated code. 

I'm doing a push now with the two methods reordered to mostly in parallel. That's helping me see where I can get some duplication out of there. Domain is not processed in a shared method. Still looking at what else can be shared.

Reply:
Worked that out. Just dealing with a Site enum instead of the whole thing.